Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -621,12 +634,12 @@ internal struct _BSONKey: CodingKey {
 }
 
 internal extension DecodingError {
-    static func _typeMismatch(at path: [CodingKey], expectation: Any.Type, reality: BSONValue?) -> DecodingError {
+    static func _typeMismatch(at path: [CodingKey], expectation: Any.Type, reality: BSONValue) -> DecodingError {
```

Comment:
Is this a candidate for `fileprivate`? AFAICT, `_typeMismatch` is only called from within `BSONDecoder.swift`.

Reply:
I don't think this is possible because the container extension (`DecodingError`) is of `internal` access level, which is weaker than `fileprivate`, so you can't make `_typeMismatch` any more stricter than `internal`.

Also, I believe this is referenced in other files too, e.g. `Document+Codable.swift` (someone can double check me), so `fileprivate` would not let us be able to use it there.